posthog/temporal/workflows/s3_batch_export.py

```diff
@@ -106,19 +106,45 @@ async def insert_into_s3_activity(inputs: S3InsertInputs):
         # when it reaches 50MB in size.
         parts: List[CompletedPartTypeDef] = []
         part_number = 1
+
         results_iterator = get_results_iterator(
             client=client,
             team_id=inputs.team_id,
             interval_start=inputs.data_interval_start,
             interval_end=inputs.data_interval_end,
         )
 
+        result = None
+
         with tempfile.NamedTemporaryFile() as local_results_file:
             while True:
                 try:
                     result = await results_iterator.__anext__()
                 except StopAsyncIteration:
                     break
+                except json.JSONDecodeError:
+                    activity.logger.info(
+                        "Failed to decode a JSON value while iterating, potentially due to a ClickHouse error"
+                    )
+                    # This is raised by aiochclient as we try to decode an error message from ClickHouse.
+                    # So far, this error message only indicated that we were too slow consuming rows.
+                    # So, we can resume from the last result.
+                    if result is None:
+                        # We failed right at the beginning
+                        new_interval_start = None
+                    else:
+                        new_interval_start = result.get("_timestamp", None)
```

Comment:
I _think_ we may need to order explicitly by `_timestamp` here to ensure that this is correct across shards, but not 100% sure about that.

Reply:
Yeah, order is 100% not guaranteed unless we ask for it...